lib/spack/spack/tengine.py

```diff
@@ -72,8 +73,12 @@ def make_environment(dirs=None):
     """Returns an configured environment for template rendering."""
     if dirs is None:
         # Default directories where to search for templates
+        builtins = spack.config.get('config:template_dirs')
+        extension_dirs = spack.config.get('config:extensions') or []
```

Comment:
I think this and the next line should be wrapped into a function in the `extension` module. Where possible, I want to avoid complicating other modules with the details of how files in Spack command extensions are arranged (even something as small as searching for a `templates` directory inside of the extension base directory).